File: c/2024/mario.c
#include <stdio.h>

void draw(int n) ;

int main(void)
{
    int height ;
    printf("Height: ") ;
    scanf("%d" , &height) ;
    while(height <= 0)
    {
        printf("Height: ") ;
        scanf("%d" , &height) ;
    }

    // printf("HEIGHT: %d\n" , height) ;
    draw(height) ;

    return 0 ;
}

void draw(int n) 
{
    for(int i = 0 ; i < n ; i++)
    {
        for(int j = (n - 1) ; j > i ; j--)
        {
            printf(" ") ;
        }
        for(int k = 0 ; k <= i ; k++)
        {
            printf("#") ;
        }

        printf(" ") ;

        for(int j = 0 ; j <= i ; j++)
        {
            printf("#") ;
        }
    
        printf("\n") ;
    }
}

File: python/2024/mario.py
def main():
    height: int = int(input("Height: "))
    while height <= 0:
        height: int = int(input("Height: "))

    draw(height)


def draw(n: int) -> None:
    for i in range(n):
        for _ in range((n - 1) , i , -1):
            print(" " , end = "")
        for _ in range(i + 1):
            print("#" , end = "")

        print("  " , end = "")

        for _ in range(i + 1):
            print("#" , end = "")

        print()

if __name__ == "__main__":
    main()
